library/gcm.c

```diff
@@ -412,8 +412,17 @@ int mbedtls_gcm_starts(mbedtls_gcm_context *ctx,
         while (iv_len > 0) {
             use_len = (iv_len < 16) ? iv_len : 16;
 
+#if defined(MBEDTLS_COMPILER_IS_GCC)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wstringop-overflow=0"
+#endif
+
```

Comment:
Doesn't `MBEDTLS_ASSUME(use_len <= 16); MBEDTLS_ASSUME(use_len < iv_len);` work here? It has the advantage of being obvious (to a human) from the line above.

Reply:
I'd prefer that, but I could not find any assumptions that would satisfy gcc.

Reply:
It seems obvious to me that this is to disable a gcc warning?

Reply:
Agreed, but `stringop-overflow` isn't an obvious warning (at least to me) - the comment in the other file also references `-O3` and includes the text of the warning message

Reply:
I think it's overkill: we don't normally provide that much detail. If someone wants to know more about the warning, they can look up this PR easily enough. I don't mind adding a comment if you think it's useful, but IMO I doubt it would be.

Reply:
As far as I'm concerned, disabling a warning around one line of code already sends the clear message that we're consciously disabling a spurious warning, and I don't think a comment would convey any more useful information.

Reply:
I do think that adding a comment along the lines of

```
gcc -O3 will generate a warning of the form 'error: writing 16 bytes into a region of size 0' for the `mbedtls_xor()` call
```

would be helpful for people coming back here in the future, without making them have to do git archaeology to figure out why the warning suppression, and whether it might no longer be relevant

Reply:
I think you are out-voted Tom, sorry, so I won't add churn for this.